tests/test_filters.py

```diff
@@ -725,6 +725,18 @@ def test_streamlines(uniform_vec):
         uniform_vec.streamlines('vectors', step_unit='not valid')
 
 
+def test_streamlines_dataset(uniform_vec):
+    vertices = np.array([[0, 0, 0], [1, 0, 0], [1, 0.5, 0], [0, 0.5, 0]])
+    source = pyvista.PolyData(vertices)
+    stream = uniform_vec.streamlines('vectors', source=source)
+    assert all([stream.n_points, stream.n_cells])
+
+    # this one fails
```

Comment:
A UniformGrid source does not work.  Using `source = pyvista.UniformGrid([3, 3, 3], [1, 1, 1], [0, 0, 0]).cast_to_unstructured_grid()` works.

Reply:
I don't think so.  It's quite likely they're expecting a certain class attribute and not finding it.  Let's cast if a uniform grid is passed.  We could even try/except cast if a dataset isn't permitted.

Reply:
This was indeed my mistake.  This source did not have any points inside the domain (on the edge or outside the domain).  I updated the test.